tests/components/russound_rio/test_config_flow.py

```diff
@@ -0,0 +1,127 @@
+from homeassistant.data_entry_flow import FlowResultType
+
+from .const import HARDWARE_MAC, MOCK_CONFIG, MODEL
+
+
+async def test_form(hass: HomeAssistant, mock_setup_entry, mock_russound) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        MOCK_CONFIG,
+    )
+
+    assert result2["type"] is FlowResultType.CREATE_ENTRY
+    assert result2["title"] == MODEL
+    assert result2["data"] == MOCK_CONFIG
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_cannot_connect(
+    hass: HomeAssistant, mock_setup_entry, mock_russound
+) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    mock_russound.connect.side_effect = TimeoutError
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        MOCK_CONFIG,
+    )
+
+    assert result2["type"] is FlowResultType.FORM
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+    # Recover with correct information
+    mock_russound.connect.side_effect = None
+    result3 = await hass.config_entries.flow.async_configure(
+        result2["flow_id"],
+        MOCK_CONFIG,
+    )
+
+    assert result3["type"] is FlowResultType.CREATE_ENTRY
+    assert result3["title"] == MODEL
+    assert result3["data"] == MOCK_CONFIG
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_no_primary_controller(
+    hass: HomeAssistant, mock_setup_entry, mock_russound
+) -> None:
+    """Test we handle no primary controller error."""
+    mock_russound.enumerate_controllers.return_value = []
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    mock_russound.enumerate_controllers.return_value = []
+
+    user_input = MOCK_CONFIG
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input
+    )
+    assert result2["type"] is FlowResultType.FORM
+    assert result2["errors"] == {"base": "no_primary_controller"}
+
+    # Recover with correct information
+    mock_russound.enumerate_controllers.return_value = [(1, HARDWARE_MAC, MODEL)]
+    result3 = await hass.config_entries.flow.async_configure(
+        result2["flow_id"],
+        MOCK_CONFIG,
+    )
+
+    assert result3["type"] is FlowResultType.CREATE_ENTRY
+    assert result3["title"] == MODEL
+    assert result3["data"] == MOCK_CONFIG
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import(hass: HomeAssistant, mock_setup_entry, mock_russound) -> None:
```

Comment:
```suggestion
async def test_import(hass: HomeAssistant, mock_setup_entry: AsyncMock, mock_russound: AsyncMock) -> None:
```
Also do this for the rest